app/controllers/transform_results_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'json'
+require 'date'
+
+##
+# Handle notifications of transform
+class TransformResultsController < Spotlight::ApplicationController
+  protect_from_forgery except: :create
+  before_action :authenticate_user!, except: :create
+
+  # This is invoked by SNS HTTP subscription
+  # rubocop:disable Metrics/AbcSize
+  def create
+    notification_msg = JSON.parse(JSON.parse(request.raw_post)['Message'])
```

Comment:
Can this be extracted into`build_notification`?  If that method also did the `DateTime` cast, could we just have `TransformResult.create(build_notification)`?